src/client/containers/ProductPage/ProductPage.Container.js

```diff
@@ -1,12 +1,86 @@
+/* eslint-disable no-console */
+/* eslint-disable radix */
 import React from 'react';
+import { useParams } from 'react-router-dom';
+import { ProductDetails } from '../../components/ProductDetails/ProductDetails.component';
+import Carousel from '../../components/Carousel/Carousel.component';
+import ButtonComponent from '../../components/Button/Button.component';
 import './ProductPage.Style.css';
+import { useFetchApi } from '../../hooks/UseFetchApi';
 
 const ProductPageContainer = () => {
+  const { id } = useParams();
+  const [product, setProduct] = React.useState({});
+  const [category, setCategory] = React.useState('');
+  const [similarProduct, setSimilarProduct] = React.useState([]);
+
+  const productData = useFetchApi(`products/${id}`);
+
+  React.useEffect(() => {
+    if (!productData.isLoading) {
+      setProduct(productData.data[0]);
+    }
+  }, [id, productData]);
+
+  const categoryData = useFetchApi(`categories/${product.category_id}`);
+
+  React.useEffect(() => {
+    if (!categoryData.isLoading) {
+      setCategory(categoryData.data[0]);
+    }
+  }, [product, categoryData]);
+
+  const similarProductData = useFetchApi(`products?category=${category.name}`);
+
+  React.useEffect(() => {
+    if (!similarProductData.isLoading) {
+      setSimilarProduct(similarProductData.data);
+    }
+  }, [similarProductData]);
+
+  const addToCartHandler = () => {
+    console.log('add to cart');
```

Comment:
It is working as expected👍. Remove console.log

Reply:
Hi Greeshma,
this is handler needs something in it .. thats why we keep console.log ...